### ISSUE TITLE
Example Checkin: Convert slang-torch/soft-rasterizer-example to idomatic slangpy

### ISSUE THREAD

**Issue by bmillsNV** [shader-slang/slangpy#76]

**Comment by jarcherNV** [comment]
I did not realise we had an issue for this. The task is mostly done already, although the current example in slangpy does not use auto-diff: https://github.com/shader-slang/slangpy-samples/tree/0a58e766f9a5783cd398c50df93924590e401096/experiments/soft-rasterizer

I will update this.

**Comment by jhelferty-nv** [comment]
Notes:
- See #75 for general notes
- The optimization part is more complex in slangpy vs pytorch. But we want to provide at least one example of how that's done in slangpy.
- Will want to use gradient descent for this example to come up with vertex positions. Needs to be added. (Use slang's autodiff backwards to get the difference. Incremental improvement is using atomics to accumulate the gradients for the verticies. Fwd is 'generate image based on these vertices'. Bkwds is saying 'how the positions of these vertices affect the image'. Chris will send some notes on how to do this.)

**Comment by jhelferty-nv** [comment]
@ccummingsNV Ping. I believe you were going to add some notes on implementing gradient descent?

**Comment by jhelferty-nv** [comment]
I've been running into problems getting this to converge. There are more details on the WIP, but I'm ending up with NaN in the gradients. The soft triangle is generate with logic with a large number of branches, so I'm attempting to simplify it so it converges before reintroducing some of that complexity.
